src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -0,0 +1,55 @@
+
+import BaseSimidCreative from '../base_simid_creative.js';
+
+/**
+ * A sample SIMID ad that shows a map of nearby locations
+ */
+
+const AdParamKeys = {
+  BUTTON_LABEL: 'buttonLabel',
+  SEARCH_QUERY: 'searchQuery',
+  MARKER: 'marker',
+};
+
+const FIND_NEAREST_TEMPLATE_TEXT = "Find Nearest ";
+const DEFAULT_BUTTON_LABEL = "Location";
+
+export default class SimidMapCreative extends BaseSimidCreative {
+
+  constructor() {
+    super();
+  }
+
+  /** @override */
+  onStart(eventData){
+    debugger;
+    super.onStart(eventData);
+    //ToDo(juliareichel@): handle invalid JSON and param errors
+    const adParams = JSON.parse(this.creativeData.adParameters);
+    const buttonLabel = adParams[AdParamKeys.BUTTON_LABEL]; 
+    const searchQuery = adParams[AdParamKeys.SEARCH_QUERY];
+    //ToDo(juliareichel@): handle case where searchQuery is undefined
+    const marker = adParams[AdParamKeys.MARKER];
+    this.specifyButtonFeatures_(buttonLabel);
+  }
+ 
+  /**
+   * Sets the text of the initial button and assigns it a click functionality.
```

Comment:
Are "initial button" and "find nearest button" the same thing?  Maybe just call them the same throughout the comment.

Reply:
yes-fixed